api/src/main/java/ai/djl/ndarray/NDArrayAdapter.java

```diff
@@ -435,79 +444,103 @@ public NDArray pow(Number n) {
     /** {@inheritDoc} */
     @Override
     public NDArray pow(NDArray other) {
-        return getAlternativeArray().pow(other);
+        return getAlternativeArray().pow(alternativeManager.from(other));
     }
 
     /** {@inheritDoc} */
     @Override
     public NDArray addi(Number n) {
-        return getAlternativeArray().addi(n);
+        NDArray array = getAlternativeArray().addi(n);
+        intern(manager.from(array));
+        return this;
     }
 
     /** {@inheritDoc} */
     @Override
     public NDArray addi(NDArray other) {
-        return getAlternativeArray().addi(other);
+        NDArray array = getAlternativeArray().addi(alternativeManager.from(other));
+        intern(manager.from(array));
+        return this;
     }
 
     /** {@inheritDoc} */
     @Override
     public NDArray subi(Number n) {
-        return getAlternativeArray().subi(n);
+        NDArray array = getAlternativeArray().subi(n);
+        intern(manager.from(array));
+        return this;
     }
 
     /** {@inheritDoc} */
     @Override
     public NDArray subi(NDArray other) {
-        return getAlternativeArray().subi(other);
+        NDArray array = getAlternativeArray().subi(alternativeManager.from(other));
+        intern(manager.from(array));
+        return this;
     }
 
     /** {@inheritDoc} */
     @Override
     public NDArray muli(Number n) {
-        return getAlternativeArray().muli(n);
+        NDArray array = getAlternativeArray().muli(n);
+        intern(manager.from(array));
+        return this;
     }
 
     /** {@inheritDoc} */
     @Override
     public NDArray muli(NDArray other) {
-        return getAlternativeArray().muli(other);
+        NDArray array = getAlternativeArray().muli(alternativeManager.from(other));
+        intern(manager.from(array));
+        return this;
     }
 
     /** {@inheritDoc} */
     @Override
     public NDArray divi(Number n) {
-        return getAlternativeArray().divi(n);
+        NDArray array = getAlternativeArray().divi(n);
+        intern(manager.from(array));
+        return this;
     }
 
     /** {@inheritDoc} */
     @Override
     public NDArray divi(NDArray other) {
-        return getAlternativeArray().divi(other);
+        NDArray array = getAlternativeArray().divi(alternativeManager.from(other));
+        intern(manager.from(array));
+        return this;
     }
 
     /** {@inheritDoc} */
     @Override
     public NDArray modi(Number n) {
-        return getAlternativeArray().modi(n);
+        NDArray array = getAlternativeArray().modi(n);
+        intern(manager.from(array));
+        return this;
     }
 
     /** {@inheritDoc} */
     @Override
     public NDArray modi(NDArray other) {
-        return getAlternativeArray().modi(other);
+        NDArray array = getAlternativeArray().modi(alternativeManager.from(other));
+        intern(manager.from(array));
```

Comment:
This conversion from the alternative manager to original manager is a wasteful of resource? Considering using ONNX GPU and PyTorch CPU at the same time. The heavier the chained NDArray operation have, the slower it will get. I would suggest the output NDArray to stay with alternative.

Reply:
This is inplace call, the return NDArray is expected to be the same as original.

Reply:
But aren't the inner content are being copied? From operation is a data copying operation from 1 engine to another

Reply:
Let's say you have a `OrtNDArray`, it contains a `OrtTensor` and `PtNDArray` as alternative array, once you called `ortArray.modi(1)`, the alternative array is updated, but `OrtTensor` is still point to the old one. If you call any operation using `OrtTensor`, the result is unexpected.